lib/ansible/modules/pip.py

```diff
@@ -663,11 +723,34 @@ def is_satisfied_by(self, version_to_test):
                 for op, ver in self._requirement.specs
             )
 
+    @property
+    def has_requirement(self) -> bool:
```

Comment:
It's nice to abstract properties like this. However, it seems inconsistent that you didn't add a docstring to this public method while added one to the others.
```suggestion
    def has_requirement(self) -> bool:
        """Compute whether the object represents complex requirement."""
```